MetadataExtractor/Formats/Jpeg/JpegMetadataReader.cs

```diff
@@ -116,8 +116,21 @@ public static DirectoryList ProcessJpegSegments([NotNull] IEnumerable<IJpegSegme
             {
                 var readerSegmentTypes = reader.SegmentTypes;
                 var readerSegments = segments.Where(s => readerSegmentTypes.Contains(s.Type));
-                if(reader is JpegDnlReader)
-                    ((JpegDnlReader)reader).ReadJpegSegments(readerSegments, directories);
+
+                // Note: JpegDnl parsing is optional. The DNL reader only defines a Height tag, 
+                //       which is redundant unless there's some problem with the regular Jpeg Height tag
```

Comment:
I'm interested to hear why you would like this to be optional. My feeling is that we should extract it if it's present. I don't think the DNL segment is that common in the wild.

Reply:
Wrong sentence structure. It's optional in the sense that nothing is parsed if jpegDirectory has an image height defined already, per Nadahar's implementation. We could change that conditional if you want but there aren't any other DNL tags that get parsed currently.

Reply:
I would prefer if we didn't special case things here.

Logic that spans directories (such as this) belongs in the as yet undesigned feature for extracting things like dates/dimensions/etc from any of the various places they may live. Discussed further in drewnoakes/metadata-extractor#10.

Reply:
Remove this special case and this PR is good to merge.

I know it'll create a difference between dotnet and java versions, but we will introduce this same behaviour in the java version too.